libs/libc/misc/lib_mutex.c

```diff
@@ -337,6 +337,7 @@ int nxmutex_unlock(FAR mutex_t *mutex)
     }
 
   DEBUGASSERT(nxmutex_is_hold(mutex));
+  ASSERT(mutex->holder == _SCHED_GETTID());
```

Comment:
but nxmutex_is_hold already do the same check.

Reply:
Sorry, based on the name of the function, I guess it's just that someone got this mutex but does not check the holder and caller is the same guy.

Reply:
it depends on your test coverage. All assert(programming error) should be fixed before your ship product.

Reply:
Thanks your reply.